docs/core/diagnostics/debug-threadpool-starvation.md

```diff
@@ -0,0 +1,312 @@
+
+- [.NET Core 6.0 SDK](https://dotnet.microsoft.com/download/dotnet) to build and run the sample app
+- [Sample web app](/samples/dotnet/samples/diagnostic-scenarios) which will demonstrate threadpool starvation behavior
+- [Bombadier](https://github.com/codesenberg/bombardier/releases) generates load for the sample web app
+- [dotnet-counters](dotnet-counters.md) to observe performance counters
+- [dotnet-stack](dotnet-stack.md) to examine thread stacks
+
+## Running the sample app
+
+1. Download the code for the [sample app](/samples/dotnet/samples/diagnostic-scenarios) and build it using the .NET SDK:
+
+  ```dotnetcli
+  E:\demo\DiagnosticScenarios>dotnet build
+  Microsoft (R) Build Engine version 17.1.1+a02f73656 for .NET
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  
+    Determining projects to restore...
+    All projects are up-to-date for restore.
+    DiagnosticScenarios -> E:\demo\DiagnosticScenarios\bin\Debug\net6.0\DiagnosticScenarios.dll
+  
+  Build succeeded.
+      0 Warning(s)
+      0 Error(s)
+  
+  Time Elapsed 00:00:01.26
+  ```
+
+1. Run the app:
+
+  ```dotnetcli
+  E:\demo\DiagnosticScenarios>bin\Debug\net6.0\DiagnosticScenarios.exe
+  info: Microsoft.Hosting.Lifetime[14]
+        Now listening on: http://localhost:5000
+  info: Microsoft.Hosting.Lifetime[14]
+        Now listening on: https://localhost:5001
+  info: Microsoft.Hosting.Lifetime[0]
+        Application started. Press Ctrl+C to shut down.
+  info: Microsoft.Hosting.Lifetime[0]
+        Hosting environment: Production
+  info: Microsoft.Hosting.Lifetime[0]
+        Content root path: E:\demo\DiagnosticScenarios
+  ```
+
+If you use a web browser and send requests to `https://localhost:5001/api/diagscenario/taskwait` you should see the response "success:taskwait" returned after about 500ms. This shows the web server is serving traffic as expected.
+
+## Observing slow performance
+
+The demo web server has several endpoints which mock doing a database request and then returning a response to the user. Each of these endpoints has a delay of approximately 500ms when serving requests one at a time but the performance is much worse when the web server is subjected to some load. Download the [Bombadier](https://github.com/codesenberg/bombardier/releases) load testing tool and observe the difference in latency when 125 concurrent requests are sent to each endpoint.
+
+```dotnetcli
+>bombardier-windows-amd64.exe https://localhost:5001/api/diagscenario/taskwait
+Bombarding https://localhost:5001/api/diagscenario/taskwait for 10s using 125 connection(s)
+[=================================================================================================================] 10s
+Done!
+Statistics        Avg      Stdev        Max
+  Reqs/sec        33.06     234.67    3313.54
+  Latency         3.48s      1.39s     10.79s
+  HTTP codes:
+    1xx - 0, 2xx - 454, 3xx - 0, 4xx - 0, 5xx - 0
+    others - 0
+  Throughput:    75.37KB/s
+```
+
```

Comment:
We need some content between these two command blocks. Even a single sentence would suffice.